File: eventhub-backend-integrated/src/main/java/com/eventhub/service/AuthService.java
package com.eventhub.service;

import com.eventhub.dto.request.*;
import com.eventhub.dto.response.AuthResponse;
import com.eventhub.dto.response.UserResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    AuthResponse refreshToken(String refreshToken);
    void logout(String refreshToken);
    UserResponse getCurrentUser();
    void forgotPassword(ForgotPasswordRequest request);
    void resetPassword(ResetPasswordRequest request);
    void verifyEmail(String token);
}

File: campus-ems-integrated/src/services/authService.js
import api from '@/lib/axios';
import EP from '@/lib/endpoints';

export const authService = {
  login:          (d)  => api.post(EP.AUTH.LOGIN, d),
  register:       (d)  => api.post(EP.AUTH.REGISTER, d),
  logout:         (rt) => api.post(`${EP.AUTH.LOGOUT}?refreshToken=${rt}`),
  forgotPassword: (e)  => api.post(EP.AUTH.FORGOT_PASSWORD, { email: e }),
  resetPassword:  (d)  => api.post(EP.AUTH.RESET_PASSWORD, d),
  verifyEmail:    (t)  => api.get(`${EP.AUTH.VERIFY_EMAIL}?token=${t}`),
  resendVerification: (e) => api.post(EP.AUTH.RESEND_VERIFICATION, { email: e }),
  getMe:          ()   => api.get(EP.AUTH.ME),
  refreshToken:   (rt) => api.post(`${EP.AUTH.REFRESH}?refreshToken=${rt}`),
};